File: tests/fixtures/javascript/match_switch.js
// Expected metrics:
// categorize: cognitive=4, cyclomatic=6, sloc=16
function categorize(x) {
    switch (x) {
        case 1:
            return "one";
        case 2:
            if (x > 0) {
                return "positive two";
            } else {
                return "two";
            }
        case 3:
            return "three";
        default:
            return "other";
    }
}

File: tests/fixtures/python/match_switch.py
# Expected metrics:
# categorize: cognitive=3, cyclomatic=6, sloc=12
def categorize(x):
    match x:
        case 1:
            return "one"
        case 2:
            if x > 0:
                return "positive two"
            return "two"
        case 3:
            return "three"
        case _:
            return "other"
